services/core/shared/downloadFile.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const https = require('https');
+const tar = require('tar');
+
+const {
+	Logger,
+	Exceptions: { NotFoundException },
+} = require('lisk-service-framework');
+
+const logger = Logger();
+
+
+const downloadZip = (snapshotUrl, directoryPath) => new Promise((resolve, reject) => {
+	https.get(snapshotUrl, (response) => {
```

Comment:
Please consider adding unit tests for the utility.

```suggestion
const downloadAndExtractTarball = (fileUrl, directoryPath) => new Promise((resolve, reject) => {
	https.get(fileUrl, (response) => {
```